tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -518,6 +519,98 @@ public void changing(LocationEvent event) {
 	for (int i = 0; i < 100; i++) browser.removeLocationListener(listener);
 }
 
+@Test
+public void test_setUrlAfterDisposed() {
+	Browser testBrowser = createBrowser(shell, SWT.NONE);
+	testBrowser.dispose();
+	assertThrows(SWTException.class, () -> testBrowser.setUrl("about:blank"));
+}
+
+@Test
+public void test_isSetUrlNotCustomTextUrlAfterSetText() {
+	String pluginPath = System.getProperty("PLUGIN_PATH");
+	testLogAppend("PLUGIN_PATH: " + pluginPath);
+	URI url;
+	// Depending on how the jUnit test is ran, (gui/maven/ant), url for local file needs to be acquired differently.
+	if (pluginPath != null) {
+		url = URI.create(pluginPath + "/data/testWebsiteWithTitle.html");
+	} else {
+		// used when ran from Eclipse gui.
+		url = URI.create(Test_org_eclipse_swt_browser_Browser.class.getClassLoader().getResource("testWebsiteWithTitle.html").toString());
+	}
+	AtomicBoolean changed = new AtomicBoolean(false);
+	browser.addLocationListener(changedAdapter(event -> {
+		if (URI.create(event.location).equals(url)) {
+			changed.set(true);
+		}
+	}));
+
+	browser.setText("Custom text");
+	browser.setUrl(url.toASCIIString());
+	shell.open();
+	assertTrue("Time Out: The Browser didn't navigate to the URL", waitForPassCondition(changed::get));
+	assertTrue("Url is wrongly considered Custom Text Url", URI.create(browser.getUrl()).equals(url)
+			&& !browser.isLocationForCustomText(browser.getUrl()));
```

Comment:
Please make two assertions out of this, to identify which of the calls leads to a failure. The URI comparison can even be replaced with an `assertEquals` to get better output in case of a failure.

Reply:
Done